src/connection.h

```diff
@@ -68,13 +82,20 @@ class Connection : public std::enable_shared_from_this<Connection>
 
 		enum { FORCE_CLOSE = true };
 
-		Connection(boost::asio::io_service& io_service,
-		ConstServicePort_ptr service_port) :
-			readTimer(io_service),
-			writeTimer(io_service),
-			service_port(std::move(service_port)),
-			socket(io_service),
-			timeConnected(time(nullptr)) {}
+		Connection(boost::asio::io_service& init_io_service,
+			ConstServicePort_ptr init_service_port) :
+			readTimer(init_io_service),
+			writeTimer(init_io_service),
+			service_port(std::move(init_service_port)),
+			socket(init_io_service) {
+			connectionState = CONNECTION_STATE_PENDING;
+			packetsSent = 0;
+			timeConnected = time(nullptr);
+			receivedFirst = false;
+			serverNameTime = 0;
+			receivedName = false;
+			receivedLastChar = false;
```

Comment:
What's the point of using the body of the constructor instead of using initialization? Also just use default values for those trivial values (false and 0)

Reply:
I would say there is no point and I hate it

Reply:
I pushed a commit and performed same tests as earlier with same results. Is it ok now?